provision/kubernetes/deploy.go

```diff
@@ -410,8 +410,16 @@ func probesFromHC(hc *provTypes.TsuruYamlHealthcheck, port int) (hcResult, error
 	if hc.TimeoutSeconds == 0 {
 		hc.TimeoutSeconds = 60
 	}
+	headers := []apiv1.HTTPHeader{}
+	for header, value := range hc.Headers {
+		headers = append(headers, apiv1.HTTPHeader{Name: header, Value: value})
+	}
 	if !hc.UseInRouter {
 		url := fmt.Sprintf("%s://localhost:%d/%s", hc.Scheme, port, strings.TrimPrefix(hc.Path, "/"))
+		headersCurl := []string{}
+		for _, header := range headers {
+			headersCurl = append(headersCurl, fmt.Sprintf("-H '%s: %s'", header.Name, header.Value))
```

Comment:
We should escape some characters from header names and values, to avoid command injection.
```suggestion
		s := fmt.Sprintf("%s: %s", header.Name, header.Value)
		headersCurl = append(headersCurl, fmt.Sprintf("-H %q", s))
```

Reply:
Ok. Method is now escaped as well.